ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -673,9 +676,13 @@ typeCheckRuleDefinition hsc pm = do
   setPriority priorityTypeCheck
   IdeOptions { optDefer = defer } <- getIdeOptions
 
-  linkables_to_keep <- currentLinkables
+  unlift <- askUnliftIO
+  let dets = TypecheckHelpers
+           { getLinkablesToKeep = unliftIO unlift $ currentLinkables
+           , getLinkables = unliftIO unlift . uses_ GetLinkable
```

Comment:
What does `unliftIO` mean here, and how is it better than `runAction`?

Reply:
We need `unliftIO` because this is going to be executed inside the GHC hook, so we can't run actions.

I used `unliftIO` instead of `runAction` so that dependencies are recorded as usual into the current rule execution.